Assets/Scripts/Framework/Train/TrainController.cs

```diff
@@ -0,0 +1,271 @@
+using UnityEngine.Splines;
+using FishNet.Object;
+using UnityEngine;
+using FishNet;
+using System;
+
+namespace DerailedDeliveries.Framework.Train
+{
+    /// <summary>
+    /// Class responsible for moving train along rails spline.
+    /// </summary>
+    [RequireComponent(typeof(TrainEngine))]
+    public class TrainController : NetworkBehaviour
+    {
+        /// <summary>
+        /// Reference to the spline line data.
+        /// </summary>
+        [field: SerializeField]
+        public SplineContainer Spline { get; private set; }
+        
+        [Header("Train Config")]
+        [SerializeField, Range(0, 1)]
+        private float _trainFrontStartTime = 0f;
```

Comment:
Default value is not needed.
```suggestion
        private float _trainFrontStartTime;
```